rs/replicated_state/src/canister_state/system_state/task_queue.rs

```diff
@@ -124,6 +124,35 @@ impl TaskQueue {
         queue
     }
 
+    pub fn from_checkpoint_v2(item: pb::TaskQueue, canister_id: &CanisterId) -> Self {
+        let queue = Self {
+            paused_or_aborted_task: item
+                .paused_or_aborted_task
+                .map(|task| task.try_into().unwrap()),
+            on_low_wasm_memory_hook_status: pb::OnLowWasmMemoryHookStatus::try_from(
+                item.on_low_wasm_memory_hook_status,
+            )
+            .unwrap()
+            .try_into()
+            .unwrap(),
+            queue: item
+                .queue
+                .into_iter()
+                .map(|task| task.try_into().unwrap())
+                .collect(),
+        };
+
+        // Because paused tasks are not allowed in checkpoint rounds when
+        // checking dts invariants that is equivalent to disabling dts.
+        queue.check_dts_invariants(
+            FlagStatus::Disabled,
+            ExecutionRoundType::CheckpointRound,
+            canister_id,
+        );
+
+        queue
+    }
```

Comment:
I think this should be a `TryFrom` following the conventions we have for converting from protobuf back into Rust structs.

So I would move it next to the `impl From<&TaskQueue> for pb::TaskQueue` as a `impl TryFrom<(pb::TaskQueue, CanisterId)> for TaskQueue`. Then deal with the errors at the deserialization boundary again following the conventions we use.

Reply:
Good point, done.